### ISSUE TITLE
If the size of the non-failed hosts is 0, stop the play

### ISSUE THREAD

**Issue by mpdehaan** [ansible/ansible#1028]
In a long playbook if all hosts fail due to PB error, it doesn't make sense to print out all the tasks in the playbook... just stop and show the summary.

In other words, if there are no non-failed hosts in the pool prior to starting a task, do not run the task.

**Comment by dagwieers** [comment]
I looked at implementing another approach. If there are no applicable hosts (either because all have failed, or because non match the 'hosts' directive) for a playbook, skip the entire playbook. I failed utterly because there are no provisions (afaict) to match hosts at the playbook level. This seems to be decided on a task level only ?

A document explaining the internal relations between the various classes and methods could be useful in this case. I feel like running into a brick wall from time to time when looking at core :-/
